### ISSUE TITLE
Adding a game on steam does not appear in GOG until a restart

### ISSUE THREAD

**Issue by ABaumher** [ABaumher/galaxy-integration-steam#39]
@Zitbvt reported this on the issue at fog's page, bringing it here. 

@don-de-marco not sure if this is addressed in the cache rewrite, or if this is something that's supposed to be handled in `tick`. I'll need to check if steam gives us a message when a new game is downloaded, installed, etc. 

Could be related to #38

**Comment by ABaumher** [comment]
@Zitbvt Apologies for the machine translation:

 Does an added game appear if you let GOG run for a little while? It may be something that's taken care of over time, or it's message we currently ignore and need fix

Est-ce qu'un jeu supplémentaire apparaît si vous laissez GOG fonctionner pendant un petit moment? Il peut s'agir de quelque chose qui a été pris en charge au fil du temps, ou c'est un message que nous ignorons actuellement et que nous devons corriger.

**Comment by don-de-marco** [comment]
> @don-de-marco not sure if this is addressed in the cache rewrite, or if this is something that's supposed to be handled in `tick`. I'll need to check if steam gives us a message when a new game is downloaded, installed, etc.
> 
> Could be related to #38

I don't think this is to be handled in `tick` because that function is called every second or so which is quite often. Maybe there's indeed a message from the Steam backend that let's us know that there's a new game available, I don't know. @Zitbvt can you please send us the plugin log file so that we can verify that?

**Comment by Zitbvt** [comment]
Comment vous adresser le log file ? Expliquez moi et je vous le déposerai
où vous allez me l'indiquer.

Cordialement.

Le sam. 1 juil. 2023 à 21:04, don-de-marco ***@***.***> a
écrit :

**Comment by don-de-marco** [comment]
Assuming that you're on Windows, the logs are in `C:\ProgramData\GOG.com\Galaxy\logs`. We need the files called `plugin-steam-...`. If you can remember, please let me know the rough time at when you added the game to your library so that I can find the relevant log lines easier

**Comment by Zitbvt** [comment]
Bonjour, j'ai 11 fichiers et 51 Mo de données concernant ces fichiers
logs... Avez vous besoin de tous ou simplement des derniers ? Dans tous les
cas je ne pourrai pas tous vous les transmettre par e-mail. Indiquez moi
comment vous le ou les adresser.

Le sam. 1 juil. 2023 à 23:42, don-de-marco ***@***.***> a
écrit :

**Comment by don-de-marco** [comment]
I need the log file from when you added the game that only appeared in Galaxy after restarting the app. I suspect there might be a message sent by Steam that we didn't parse correctly which resulted in the game only showing up after the next full sync (after you closed and re-opened Galaxy).

You can just attach the files to this issue by dragging them into the text box when typing your answer.

**Comment by Zitbvt** [comment]
Bonsoir, voici mon fichier log.
Merci pour votre travail.

Cordialement.
[plugin-steam-ca27391f-2675-49b1-92c0-896d43afa4f8.log](https://github.com/ABaumher/galaxy-integration-steam/files/11951925/plugin-steam-ca27391f-2675-49b1-92c0-896d43afa4f8.log)
